Modules/Loadable/Markups/MRML/vtkCurveMeasurementsCalculator.cxx

```diff
@@ -0,0 +1,352 @@
+vtkCollection* vtkCurveMeasurementsCalculator::GetMeasurements()
+{
+  return this->Measurements.GetPointer();
+}
+
+//----------------------------------------------------------------------------
+int vtkCurveMeasurementsCalculator::FillInputPortInformation(
+  int port, vtkInformation* info)
+{
+  if (port == 0)
+    {
+    info->Set(vtkAlgorithm::INPUT_REQUIRED_DATA_TYPE(), "vtkPolyData");
+    }
+  else
+    {
+    vtkErrorMacro("Cannot set input info for port " << port);
+    return 0;
+    }
+
+  return 1;
+}
+
+//------------------------------------------------------------------------------
+int vtkCurveMeasurementsCalculator::RequestData(
+  vtkInformation* vtkNotUsed(request),
+  vtkInformationVector** inputVector,
+  vtkInformationVector* outputVector)
+{
+  vtkInformation* inInfo = inputVector[0]->GetInformationObject(0);
+  vtkPolyData* inputPolyData = vtkPolyData::SafeDownCast(
+    inInfo->Get(vtkDataObject::DATA_OBJECT()));
+
+  if (!inputPolyData)
+    {
+    return 1;
+    }
+
+  // vtkPoints* inputPoints = inputPolyData->GetPoints();
+  // if (!inputPoints)
+    // {
+    // return 1;
+    // }
+
+  vtkInformation* outInfo = outputVector->GetInformationObject(0);
+  vtkPolyData* outputPolyData = vtkPolyData::SafeDownCast(
+    outInfo->Get(vtkDataObject::DATA_OBJECT()));
+
+  outputPolyData->CopyStructure(inputPolyData);
+  outputPolyData->GetPointData()->PassData(inputPolyData->GetPointData());
+  outputPolyData->GetCellData()->PassData(inputPolyData->GetCellData());
+  outputPolyData->GetFieldData()->PassData(inputPolyData->GetFieldData());
+
+  if (this->CalculateCurvature)
+    {
+    if (!this->CalculatePolyDataCurvature(outputPolyData))
+      {
+      return 0;
+      }
+    }
+  else
+    {
+    outputPolyData->GetPointData()->RemoveArray("Curvature");
+    }
+
+  // If interpolating measurements is enabled then go through measurements, and
+  // interpolate those that contain control point data and are enabled
+  if (this->InterpolateControlPointMeasurement)
+    {
+    this->InterpolateControlPointMeasurementToPolyData(outputPolyData);
+    }
+  else
+    {
+    //TODO: Remove arrays? Prefix their names with "Interpolated:"?
+    }
+
+  outputPolyData->Squeeze();
+  return 1;
+}
+
+//------------------------------------------------------------------------------
+bool vtkCurveMeasurementsCalculator::CalculatePolyDataCurvature(vtkPolyData* polyData)
+{
+  if (polyData == nullptr)
+    {
+    return false;
+    }
+  if (polyData->GetNumberOfPoints() == 0 || polyData->GetNumberOfLines() == 0)
+    {
+    // vtkErrorMacro("CalculatePolyDataCurvature: No points or lines in input poly data");
+    return false;
+    }
+
+  // Note: This algorithm has been ported from CurveMaker
+  //       https://github.com/tokjun/CurveMaker/blob/master/CurveMaker/CurveMaker.py
+  // TODO: This routine does not consider a closed loop. If a closed loop is specified,
```

Comment:
Could you add closed curve computation, too? We would just need to wrap around at both ends (to compute curvature at one end, you would need to use point(s) from the other end).